### ISSUE TITLE
Windows -n routing option, wrong gateway

### ISSUE THREAD

**Issue by leonardospeca** [ntop/n2n#528]
Hi, I have a problem with the **-n** option in Windows, this is the situation:
- I'm IP **192.168.1.34** with gateway **192.168.1.1**
- I'm connecting with ip **10.254.254.11**, to an edge with ip **10.254.254.10** and two subnet: **10.10.10.0** and **10.123.123.0**
- I'm using options **-n10.10.10.0/24:10.254.254.10** and **-n10.123.123.0/24:10.254.254.10**
The fact is that, I can ping 10.254.254.10 successfully, but I can't ping anything in the subnet 10.10.10.0 or 10.123.123.0

To solve I have just to run this command:
`route change 10.10.10.0 mask 255.255.255.0 10.254.254.10`, I can successfully go in subnet 10.10.10.0

So, why under the interface column, when I use the -n option, there isn't the n2n network adapter but the pc one?

**Before:** (not working)
![ip2](https://user-images.githubusercontent.com/26072174/101384632-faef6e80-38ba-11eb-9709-cbf6051e3818.png)
**After:** (working 10.10.10.0)
![ip3](https://user-images.githubusercontent.com/26072174/101385090-97b20c00-38bb-11eb-98de-57628cb5c3a0.png)

**Comment by leonardospeca** [comment]
In addition, when I close the connection, those rules don't go away, because the interface **192.168.1.34** is still there of corse.
If I change those two rules, it automatically give them the correct interface **10.254.254.11** (My Ip in the vpn) and when the connection close, those rules go away correctly

**Comment by Logan007** [comment]
Regarding Windows (and MacOS), I received a similar report from another user in a private discussion basically saying that `-n` seemed broken on Windows and MacOS. Only solution I have heard of so far requires to manually add `route` commands in a `.bat` file after calling `edge …-r…`.

Perhaps, @emanuele-f who is the original author of the route control code has got an idea on how to make this part work on Windows (and MacOS).

**Comment by ChenJun-github-lab** [comment]
I also had this problem. After checking the source code, I found that the reason for this error was that the tuntap device was not ready when the add router was executed, so there was no suitable interface to match the gateway.

I put the program edge_ init_ Routes() is executed from edge_ Init() is modified after the TUNTAP device is set.

<edge_utils.c>
![image](https://user-images.githubusercontent.com/49386141/111904847-d98a4b00-8a83-11eb-8878-49cad8cb0cd7.png)

<edge.c>   （Note that  edge_ init_ Routes() becomes a non static method）
![image](https://user-images.githubusercontent.com/49386141/111904867-ead35780-8a83-11eb-823c-bdec9bc6ffda.png)

But there are other problems after that. In the process of debugging, I used the ipconfig command of CMD to find that Windows system seems to need some time to test the TUNTAP device after it is set and started. So I added the sleep (5) command after the tuntap device is set.

After recompiling and running, my edge.exe  It's OK to use the - N instruction in.


**It's just a temporary solution.**

I think a better way is to regularly query whether the TUNTAP device can be used after setting it (when it is not ready, IP will be displayed after (test) using ipconfig in CMD )
